### PR TITLE
Add Datepicker to Sidebar

### PR DESCRIPTION
**Pull request by CamronStaley**
## What changes are proposed in this pull request?

Add Datepicker to Sidebar

## How is this patch tested? If it is not, please explain why.

WORK IN PROGRESS TESTING NEED TO CHECK ALL EDGE CASES INCLUDING BUT NOT LIMITED TO:

- Different app timezones
- Larger date ranges
- Multiple dates at the same time
- A group of dates (is this even possible idk??)

### Is this a user-facing change that should be mentioned in the release notes?

-   [ ] No. You can skip the rest of this section.
-   [x] Yes. Give a description of this change to be included in the release
        notes for FiftyOne users.

Date and DateTime are no longer sliders but are instead a popup date picker.

### What areas of FiftyOne does this PR affect?

-   [x] App: FiftyOne application changes
-   [ ] Build: Build and test infrastructure changes
-   [ ] Core: Core `fiftyone` Python library changes
-   [ ] Documentation: FiftyOne documentation changes
-   [ ] Other

## Release Notes

- **New Features**
	- Introduced a flexible `BasicDatePicker` component with enhanced configuration options, including date-time selection and range limits.
	- Added a new `DateFieldFilter` component for improved date filtering capabilities.
	- Implemented a `Boxes` component for managing date range selections visually.

- **Bug Fixes**
	- Updated filtering logic to correctly utilize the `DateFieldFilter` for date-related fields instead of numeric filters.

- **Chores**
	- Simplified import paths for the `DateFieldFilter` component for easier access.

### CODE REVIEW

File: app/packages/components/src/components/DatePicker/DatePicker.tsx [hunk 1]
```diff
@@ -1,39 +1,58 @@
...
+          value={value}
+          onChange={(newValue) => {
+            onChange(newValue);
+          }}
+          minDate={minDate}
+          maxDate={maxDate}
+          format={format}
+        />
+      ) : (
+        <DatePicker
```
Comment (minhtuev) [inline app/packages/components/src/components/DatePicker/DatePicker.tsx:43]: nice!

File: app/packages/core/src/components/Filters/DateFieldFilter/Boxes.tsx [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+  display: flex;
+  justify-content: space-between;
+  column-gap: 1rem;
+  background: ${({ theme }) => theme.background.level2};
+  border: 1px solid var(--fo-palette-divider);
+  border-radius: 2px;s
+  color: ${({ theme }) => theme.text.secondary};
+  margin-top: 0.25rem;
+  padding: 0.25rem 0.5rem;
+`;
```
[the remark this answers is not in the document]
Comment (minhtuev) [reply app/packages/core/src/components/Filters/DateFieldFilter/Boxes.tsx:26]: The AI is right :) @CamronStaley

**Review by minhtuev — APPROVED** [review]
LGTM 👍

**Comment by CamronStaley** [comment]
Closing pull request because we decided to change direction for now. This can be reopened in the future if we'd like but for now we are going to just hide range filters that are in a group within the sidebar until the user presses a button to reveal it.

**Pull request closed without merging**